test/StableJoeStaking.test.js

```diff
@@ -672,6 +697,29 @@ describe("Stable Joe Staking", function () {
       );
     });
 
+    it("should allow EOAs with Smol Joes to be exempt from paying deposit fee", async function () {
+      await this.stableJoeStaking
+        .connect(this.dylan)
+        .deposit(ethers.utils.parseEther("100"));
```

Comment:
There doesn't look to be a bug there but for fuller test coverage you should also deposit from the vault before transferring the ERC721 to it.